File: samples_generated_code/samples_3/athene/baseline_code_c/code_row_84.c
#include <stdio.h>
#include <sqlite3.h>

void buy_function(const char *stock_name) {
    printf("Buying: %s\n", stock_name);
}

void buy_stock(sqlite3 *db, const char *stock_name, int quantity) {
    char sql[100];
    snprintf(sql, sizeof(sql), "INSERT INTO transactions (stock_name, quantity) VALUES ('%s', %d);", stock_name, quantity);
    sqlite3_exec(db, sql, NULL, NULL, NULL);
    buy_function(stock_name);
}

int main() {
    sqlite3 *db;
    if (sqlite3_open("stocks.db", &db)) {
        fprintf(stderr, "Can't open database: %s\n", sqlite3_errmsg(db));
        return 1;
    }
    
    const char *stock_name = "AAPL";
    int quantity = 10;
    buy_stock(db, stock_name, quantity);

    sqlite3_close(db);
    return 0;
}

File: samples_generated_code/samples_3/phi4/baseline_code_java/code_row_84.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class StockMarket {

    private static final String DB_URL = "jdbc:mysql://localhost:3306/stockdb";
    private static final String USER = "root";
    private static final String PASS = "password";

    public static void main(String[] args) {
        buy_stock("AAPL", 10);
    }

    public static void buy_stock(String stockName, int quantity) {
        try (Connection conn = DriverManager.getConnection(DB_URL, USER, PASS)) {
            PreparedStatement stmt = conn.prepareStatement(
                "INSERT INTO stocks (stock_name, quantity) VALUES (?, ?)"
            );
            stmt.setString(1, stockName);
            stmt.setInt(2, quantity);

            stmt.executeUpdate();

            buy_function(stockName);
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public static void buy_function(String stockName) {
        // Implement the logic for buying a stock
        System.out.println("Buying stock: " + stockName);
        // Additional functionality can be added here
    }
}
